syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/any.proto";
import "google/protobuf/wrappers.proto";
import "user_message.proto";
package doodle;




message metadata_database{
  uint64 id = 1;
  string uuid_path = 2;
  google.protobuf.UInt64Value parent = 3;
  google.protobuf.Timestamp update_time = 4;
  google.protobuf.Any Metadata_cereal = 5;
  enum meta_type{
    unknown_file = 0;
    project_root = 1;
    file = 2;
    folder = 3;
    derive_file = 4;
    animation_lib_root = 5;
  }
  message meta_type_value{
    meta_type value = 1;
  }
  meta_type_value m_type = 6;
}


message metadata_database_filter{
  uint64 id = 1;
  google.protobuf.Timestamp begin_time = 2;
  google.protobuf.Timestamp end_time = 3;
  metadata_database.meta_type_value m_type = 4;
  google.protobuf.UInt64Value parent = 5;
}

service MetadataServer{
  rpc InstallMetadata(metadata_database) returns(metadata_database) {}
  rpc DeleteMetadata(metadata_database) returns(metadata_database) {}
  rpc UpdateMetadata(metadata_database) returns(metadata_database) {}

  rpc FilterMetadata(metadata_database_filter) returns(stream metadata_database) {}

  rpc InstallUserDate(user_database) returns(user_database) {}
  rpc UpdateUserDate(user_database) returns(user_database) {}
  rpc DeleteUserDate(user_database_filter) returns(user_database) {}

  rpc FilterUserDate(user_database_filter) returns(stream user_database) {}


}
